server/test/endpoints/document.spec.js

```diff
@@ -0,0 +1,710 @@
+import supertest from 'supertest';
+import { expect } from 'chai';
+import findIndex from 'lodash/findIndex';
+import app from '../../../server';
+import database from '../../models';
+import SpecHelper from '../helpers/SpecHelper';
+import SeedHelper from '../helpers/SeedHelper';
+
+const client = supertest.agent(app);
+
+describe('Documents:', () => {
+  const adminUser = { ...SpecHelper.validAdminUser };
+  const regularUser1 = SpecHelper.generateRandomUser(2);
+  const regularUser2 = SpecHelper.generateRandomUser(2);
+  const regularUser3 = SpecHelper.generateRandomUser(2);
+  const userWithoutDocument = SpecHelper.generateRandomUser(2);
+  before((done) => {
+    // generate regular and admin user data after clearing database
+    SeedHelper.init()
+    .then(() => {
+      // login admin
+      client.post('/users/login')
+      .send({
+        email: adminUser.email,
+        password: adminUser.password
+      })
+      .end((error, response) => {
+        // set admin token and id
+        adminUser.token = response.body.activeToken;
+        adminUser.id = response.body.userId;
+        // fetch regularUser1 details
+        client.post('/users')
+        .send(regularUser1)
+        .end((error1, response1) => {
+          regularUser1.token = response1.body.activeToken;
+          regularUser1.id = response1.body.userId;
+          // fetch regularUser2 details
+          client.post('/users')
+          .send(regularUser2)
+          .end((error2, response2) => {
+            regularUser2.token = response2.body.activeToken;
+            regularUser2.id = response2.body.userId;
+            client.post('/users')
+            .send(regularUser3)
+            .end((error3, response3) => {
+              regularUser3.token = response3.body.activeToken;
+              regularUser3.id = response3.body.userId;
+              client.post('/users')
+              .send(userWithoutDocument)
+              .end((error4, response4) => {
+                userWithoutDocument.token = response4.body.activeToken;
+                userWithoutDocument.id = response4.body.userId;
+                done();
+              });
+            });
+          });
+        });
+      });
+    });
+  });
+
+  after((done) => {
+    database.sequelize.sync({ force: true })
+    .then(() => {
+      done();
+    });
+  });
+
+  describe('Post', () => {
+    it('should create a new Document if title and content are provided',
+    (done) => {
+      const publicDocument = SpecHelper.generateRandomDocument('public');
+      client.post('/documents')
+      .send(publicDocument)
+      .set({ 'xsrf-token': regularUser1.token })
+      .end((error, response) => {
+        expect(response.status).to.equal(201);
+        expect(response.body.createdAt).to.not.equal(undefined);
+        expect(response.body.title).to.equal(publicDocument.title);
+        done();
+      });
+    });
+
+    it(`should override docuement id and go ahead to create a
```

Comment:
In this test suite, how are you sure that the document Id is overridden.  You don't seem to actually be testing for it.